vcluster/deploy/topologies/air-gapped.mdx

```diff
@@ -16,241 +16,237 @@ import FipsConfig from '../../_partials/deploy/fips-config.mdx';
+  :::
 
-chmod +x ./download-images.sh
-chmod +x ./push-images.sh
-```
+  </Step>
+  <Step>
+
+  Run `push-images.sh` to upload all required images to your private registry:
+  
+  ```bash title="Push images to private registry"
+  
+  ./push-images.sh --registry ${REGISTRY} 
+  ```
 
   </Step>
   <Step>
+  **Optional**: If you want to deploy vCluster with a different Kubernetes version, download the `images-optional.txt` file. It contains additional container images required for that version, which you'll need to pull and push to your private registry.
+    
+    :::note
+    The `images.txt` contains multiple Kubernets versions and distributions. You can edit the file and remove the images for the unused versions and distributions. 
+    :::
 
-Run `download-images.sh` to download all images locally:
+    ```bash title "Pull, package, and push all optional images"
+    wget https://github.com/loft-sh/vcluster/releases/download/v${VCLUSTER_VERSION}/images-optional.txt
+    ./download-images.sh --image-list images-optional.txt
+    ./push-images.sh --registry ${REGISTRY}
+    ```
+    
+  </Step>
+</Flow>
 
-```bash title="Download images"
-./download-images.sh --image-list images.txt
-```
+## Populate the vCluster Helm chart to a private registry {#populate-vCluster-Helm-chart-private-registry}
 
-:::tip
-This creates a tarball with all the images and push them to your private registry.
-:::
+You need to push the vCluster Helm chart to your OCI-compliant private registry. 
 
-  </Step>
+### Prerequisites 
+
+- Access to the internet
+- Ability to push to your OCI-compliant private registry
+- `curl` installed
+- `helm` installed: Helm v3.10+
+
+### Download and push the vCluster Helm chart {#download-push-vCluster-Helm-chart}
+
+<Flow id="offline-helm-chart">
   <Step>
 
-Run `push-images.sh` to push all downloaded images to your private registry:
+  **Optional**: If you haven’t already set the environment variables, set them now before continuing.
 
-```bash title="Push images to private registry"
-./push-images.sh --registry ${REGISTRY}
-```
+   ```bash title="Export environment variables"
+   export VCLUSTER_VERSION=0.25.0 # Replace with the desired version
+   export REGISTRY=ecr.io/myteam # This should be a prefix; do not include any image paths
+   ```
+  </Step>
+  <Step>
 
-:::info
-vCluster prepends the image registry to all images used by vCluster, such as syncer and Kubernetes. For example, `registry.k8s.io/kube-apiserver:v1.30.2` becomes `my-private-registry:5000/vcluster/kube-apiserver:v1.30.2`.
-:::
+  ```bash title="Download the vCluster Helm chart
+  curl -O https://charts.loft.sh/charts/vcluster-"${VCLUSTER_VERSION}".tgz
+  ```
+  </Step>
 
-    </Step>
+  <Step>
 
+  ```bash title="Push your chart to your private registry"
+  helm chart save ./vcluster-${VCLUSTER_VERSION}.tgz ${REGISTRY}/vcluster:v${VCLUSTER_VERSION}
+  helm chart push ${REGISTRY}/vcluster:v${VCLUSTER_VERSION}
```

Comment:
It should work with just `helm pull` and `helm push`. Also, you need to include the oci protocol:
```
helm pull vcluster --repo https://charts.loft.sh --version ${VCLUSTER_VERSION}
helm push vcluster-${VCLUSTER_VERSION}.tgz oci://${REGISTRY}/charts
```

Reply:
Do we expect the registry to have a `/charts` repo? We haven't made that a pre-req.